### DISCUSSION TITLE
Render to Frontend with template Markdown or Code

### DISCUSSION THREAD

**Discussion by tqt97** [statamic/cms#8599] [Q&A] [answered]
Hi there, I'm a newbie. 
I wanna to render Code Field or Markdown Field to Frontend. 

**Description:**

In backend i use fields

![image](https://github.com/statamic/cms/assets/32692426/ee5687fd-4800-483d-8f73-aa72a280b6fc)

Result in frontend 

![image](https://github.com/statamic/cms/assets/32692426/f05acef6-6c25-4853-8cc5-db47655668e2)

and I want to show in frontend with template same bellow

![image](https://github.com/statamic/cms/assets/32692426/903e4bed-a048-47b8-a79d-35cba2287608)

or same bellow. It maybe same Torchlight plugin

![In frontend](https://github.com/statamic/cms/assets/32692426/0fa9e031-4a2b-4d38-85f2-fbbac2604916)

Please help me. Thanks so much

**Comment by duncanmcclean** [comment, marked as answer]
Hey 👋🏻

I've written a blog post about setting up [Torchlight and Bard in Statamic](https://duncanmcclean.com/torchlight-statamic-bard), if you're interested.

If you'd prefer to use something local (eg. without the need for sending API requests for code embeds), you could look at implementing something like Shiki. 

Here's a very quick tag I've created for Shiki:

```php
<?php

namespace App\Tags;

use Illuminate\Support\Facades\Cache;
use Spatie\ShikiPhp\Shiki as ShikiPhp;
use Statamic\Tags\Tags;

class Shiki extends Tags
{
    /**
     * The {{ shiki :code="code" :language="language" }} tag.
     *
     * @return string|array
     */
    public function index()
    {
        return Cache::rememberForever('ShikiHighlight::'.md5($this->params->get('code').$this->params->get('language')), function () {
            return ShikiPhp::highlight(
                code: $this->params->get('code'),
                language: $this->params->get('language'),
                theme: 'slack-ochin',
            );
        });
    }
}
```

```antlers
{{ shiki :code="code" :language="language" }}
```

**Reply by tqt97** [reply]
Thanks a lot. It's very useful.
